binary_gentoo/internal/cli/tree_diff.py

```diff
@@ -3,11 +3,26 @@
 
 import filecmp
 import os
+import re
 import sys
 from argparse import ArgumentParser
 
-from ..reporter import exception_reporting
 from ._parser import add_version_argument_to
+from ..reporter import exception_reporting
+
+keywords_pattern = re.compile('KEYWORDS="(?P<keywords>.*?)"')
+
+
+def _get_relevant_keywords_set_for(ebuild_content, expected_keywords):
+    try:
+        ebuild_keywords = keywords_pattern.search(ebuild_content).group('keywords')
+    except AttributeError:
+        # if the KEYWORDS variable is not found in the ebuild, we will assume the ebuild needs to be listed
+        ebuild_keywords = expected_keywords
+    expected_keywords_set = set(expected_keywords.split(" "))
```

Comment:
With picky hat on: While not totally out of place, I would somehow expect this to be done outside the function and make the function signature take a set or some pre-split iterable. Can probably be argued, but maybe you felt something similar about it.

Reply:
You're right. Since the ebuild keywords needed the same split, I kept it in the same place but it is indeed more elegant if the function accepts some iterable instead of a space-separated keywords string.

Reply:
I have modified so it accepts an iterable, which it then converts to a set. But since a string is an iterable too, I have added a check to see if it's a string. If it is, it does a split on spaces. So basically it's a bit the best of both worlds now.